src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3719,6 +3720,31 @@ int sockinfo_tcp::getsockopt_offload(int __level, int __optname, void *__optval,
 				errno = EINVAL;
 			}
 			break;
+		case TCP_INFO:
+			if (*__optlen >= sizeof(int)) {
+				struct tcp_info *ti = (struct tcp_info*)__optval;
+				int cnvt_to_tcpi_state[TCP_CLOSING] = { \
+					TCP_CLOSE, TCP_LISTEN, TCP_SYN_SENT, TCP_SYN_RECV,
+					TCP_ESTABLISHED, TCP_FIN_WAIT1, TCP_FIN_WAIT2, TCP_CLOSE_WAIT,
+					TCP_CLOSING, TCP_LAST_ACK, TCP_TIME_WAIT,
+				};
+				/* currently, limited support */
+				ti->tcpi_state = cnvt_to_tcpi_state[m_pcb.private_state];
+				ti->tcpi_retransmits = m_pcb.nrtx;
+				ti->tcpi_rto = m_pcb.rto;
+				ti->tcpi_snd_wscale = m_pcb.snd_scale;
+				ti->tcpi_rcv_wscale = m_pcb.rcv_scale;
+				ti->tcpi_snd_mss = m_pcb.mss;
+				ti->tcpi_rcv_mss = m_pcb.mss;
+				ti->tcpi_advmss = m_pcb.advtsd_mss;
+				ti->tcpi_snd_cwnd = m_pcb.cwnd;
+				ti->tcpi_snd_ssthresh = m_pcb.ssthresh;
+				ti->tcpi_unacked = m_pcb.unacked->seqno;
+				ret = 0;
+			} else {
+				errno = EINVAL;
+			}
+			break;
```

Comment:
Is this related to the fix? if not we need to split it into two commits.

Reply:
This code related to the fix but only used in PD.  I agree this code should be removed.